File: nightgrass/src/main/java/com/goodfeel/nightgrass/dto/CartItemDto.java
package com.goodfeel.nightgrass.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Getter @Setter @AllArgsConstructor @NoArgsConstructor
public class CartItemDto {

    private Long itemId;
    private Long cartId;
    private Long productId;
    private String productName;
    private int quantity;
    private String properties;
    private BigDecimal price;

}

File: nightgrass/src/main/kotlin/com/goodfeel/nightgrass/dto/CartItemDto.kt
package com.goodfeel.nightgrass.dto

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import java.math.BigDecimal

data class CartItemDto(
    val itemId: Long? = null,
    val productId: Long,
    val imageUrl: String,
    val productName: String,
    val description: String,
    val quantity: Int,
    val properties: String? = null,
    var formattedProperties: Map<String, String> = mapOf(),
    val price: BigDecimal,
    val discountedPrice: BigDecimal,
    val formattedPrice: String? = null,
    val isSelected: Boolean = true
) {
    companion object {
       private val objectMapper = ObjectMapper()
    }

    fun processProperties() {
        formattedProperties = properties?.let {
            objectMapper.readValue(it, object : TypeReference<Map<String, String>>() {})
        } ?: mapOf()
    }
}
